### ISSUE TITLE
Solr chart - JVM Mem Settings using "javaMem" is not working

### ISSUE THREAD

**Issue by rmuehlbauer** [bitnami/charts#6812]
**Which chart**:
Bitnami/Solr Version 1.0.1
Docker container used was bitnami/solr:8.9.0-debian-10-r5

**Describe the bug**
Setting the "javaMem" parameter in values.yaml, does not affect the memory setting of the JVM 

**To Reproduce**
1. Set "javaMem" in values.yaml to e.g. "-Xms512m -Xmx2048m"
2. helm upgrade/install your solr installation
3. check the running JVM for its memory parameters using "ps -ef" inside the running solr container - the JVM is running with different mem settings - here is only interesting part of the "ps -ef" output: "/opt/bitnami/java/bin/java -server -Xms512m -Xmx512m -XX:+UseG1GC -XX:+PerfDisableSharedMem....."
 

**Expected behavior**
JVM should use given javaMem values.

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.6.2", GitCommit:"ee407bdf364942bcb8e8c665f82e15aa28009b71", GitTreeState:"dirty", GoVersion:"go1.16.5"}
```

- Output of `kubectl version`:

```
kubectl version
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2", GitCommit:"092fbfbf53427de67cac1e9fa54aaa09a28371d7", GitTreeState:"clean", BuildDate:"2021-06-16T12:52:14Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.11-gke.2100", GitCommit:"6c93c85d1c906a6502383799c8e56606e10aeb96", GitTreeState:"clean", BuildDate:"2021-06-11T09:24:24Z", GoVersion:"go1.15.12b5", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.21) and server (1.19) exceeds the supported minor version skew of +/-1
```

**Additional context**
The "javaMem" values given in values.yaml are ending up in an ENV variable called SOLR_JAVA_MEM - I think thats fine.
```
 I have no name!@dev-solr-2:/$ env | grep -i mem
SOLR_JAVA_MEM=-Xms512m -Xmx2048m
I have no name!@dev-solr-2:/$
```
The problem seems to be, that the bitnami/solr docker container itself just does not use the env variable for mem settings.
I think the script responsible for doing solr related environment stuff is /opt/bitnami/scripts/solr-env.sh and it just does not care about the SOLR_JAVA_MEM environment variable...
How is this intended to work?
How can I tweak JVM Mem Settings?

**Comment by migruiz4** [comment]
Hi @rmuehlbauer,

Thank you for reporting this issue.

The problem is caused by `SOLR_HEAP` being set to 512m by default. SOLR_HEAP has priority over SOLR_JAVA_MEM, causing it to be ignored.

Modifying `solr-env.sh` to set SOLR_JAVA_MEM instead of SOLR_HEAP should fix the issue:
```diff

- export SOLR_HEAP="${SOLR_HEAP:-512m}"

+ export SOLR_HEAP="${SOLR_HEAP:-}"
+ export SOLR_JAVA_MEM="${SOLR_JAVA_MEM:--Xms512m -Xmx512m}"
```
 
A new version of the Solr Docker image including the fix will be released soon.

**Comment by rmuehlbauer** [comment]
Hi @migruiz4,

thank you very much for your fast reaction - your explanation sounds plausible!
I'm looking forward to test the updated docker image.

**Comment by rmuehlbauer** [comment]
hm, maybe I was wrong....
There is already a parameter called "heap" available in the solr chart.
If I set this to e.g. "1024m" the running solr docker container now hat 1024mb ob heap available:
`/opt/bitnami/java/bin/java -server -Xms1024m -Xmx1024m`

I'm not sure if you need to change anything at all - for now it looks like this "heap" parameter exactly does what I want...

**Comment by migruiz4** [comment]
You weren't wrong at all! 
Your issue indeed helped us improve the Solr image.

Solr has two env Variables to configure Heap:
- SOLR_HEAP (chart value `heap`). Configures the minimum and maximum memory heap to the same value.
- SOLR_JAVA_MEM (chart value `javaMem`). Allows more advanced heap settings, such as providing different minimum and maximum.

As described in `solr.in.sh`, SOLR_JAVA_MEM will be ignored when SOLR_HEAP.
We were setting SOLR_HEAP=512m by default, so SOLR_JAVA_MEM(`javaMem`) was being ignored.

With the new image, both settings can be used, but keep in mind SOLR_HEAP will have priority if both are set simultaneously.

The image hasn't been released yet, I will let you know as soon as it is available.

**Comment by rmuehlbauer** [comment]
If so - I'm glad I could help :-)

At the moment I was at least able to set a somewhat higher heap for my solr instances, but I will keep in mind SOLR_HEAP has a higher priority and use SOLR_JAVA_MEM instead, once your new image has been released.

Thank you very much.

**Comment by migruiz4** [comment]
Hi @rmuehlbauer,

A new version of bitnami/solr has been released: 8.9.0-debian-10-r10.
This image includes the changes to the SOLR_HEAP and SOLR_JAVA_MEM env variables.

Using this image you should be able to configure `javaMem` with values such as '-Xms512m -Xmx2048m' as in the issue description.

If you find any further issues please let us know.

**Comment by rmuehlbauer** [comment]
Hi @migruiz4,

thank you very much for the adaption - for me it is now working exactly as expected.
Both SOLR_HEAP and SOLR_JAVA_MEM env variables are now respected in the way they should.
